R/pkg/R/SQLContext.R

```diff
@@ -90,19 +90,18 @@ createDataFrame <- function(sqlContext, data, schema = NULL, samplingRatio = 1.0
       if (is.null(schema)) {
         schema <- names(data)
       }
-      n <- nrow(data)
-      m <- ncol(data)
-      # get rid of factor type
-      dropFactor <- function(x) {
+      # get rid of factor type and adjust for lists
+      cleanCols <- function(x) {
         if (is.factor(x)) {
           as.character(x)
+        } else if(is.list(x) && !is.environment(x)) {
```

Comment:
Seems the code only handle the special case of list here. What about an environment? (As an environment is for a Map type )